### PR TITLE
Add Bazel remote cache to package build

### PR DESCRIPTION
**Pull request by GMNGeoffrey**
This uses a GCS bucket as a remote cache partitioned by docker image
name and digest.

This can be used for other builds as well using the same process.
Speedups here are not spectacular, but it helps a bit.

Tested:
https://github.com/openxla/openxla-pjrt-plugin/actions/workflows/build_packages.yml
Run 1 was writing the cache only and run 2 was with caching.

### CODE REVIEW

File: .github/workflows/build_packages.yml [hunk 1]
```diff
@@ -25,6 +25,15 @@ jobs:
...
         uses: actions/checkout@e2f20e631ae6d7dd3b768f56a5d2af784dd54791 # v2.5.0
 
+      - id: "gcp-auth"
+        name: "Authenticating to Google Cloud"
+        if: github.event_name != 'pull_request'
+        uses: "google-github-actions/auth@v1"
+        with:
+          token_format: "access_token"
+          credentials_json: "${{ secrets.IREE_OSS_GITHUB_RUNNER_BASIC_TRUST_SERVICE_ACCOUNT_KEY }}"
+          create_credentials_file: true
```
Comment (stellaraccident) [inline .github/workflows/build_packages.yml:35]: Where does the credentials file go? I think we need to do something to make sure it goes in the docker container, no?

**Review by stellaraccident — COMMENTED** [review]
Looks like the auth file doesn't exist in the docker container?

File: .github/workflows/build_packages.yml (hunk 1 above)
Comment (GMNGeoffrey) [reply .github/workflows/build_packages.yml:35]: It doesn't on presubmit, which is intentional. I didn't realize `--google_default_credentials` would error in this case. The file is placed in `GITHUB_WORKSPACE` so it gets mounted in the container with everything else.

Comment (stellaraccident) [reply .github/workflows/build_packages.yml:35]: So for my own understanding, we write in post-submit and pre-submits only read, correct?

**Review by stellaraccident — APPROVED** [review]

**Pull request merged**